### ISSUE TITLE
Make the error messages include the location of the cause

### ISSUE THREAD

**Issue by boosh** [facebook/relay#304]
I'm getting this error in the following fragment:
```
  fragments: {
    viewer: () => Relay.QL`
      fragment on Viewer {
        ${StatusMutation.getFragment('viewer')}
        calendar(id: $meetingId) {
          id
        }
      }
    `,
  },
```

 I've tried changing my use of `getFragment` to all of the following as per the error, none of which have worked:

```
${StatusMutation.getFragment('viewer')}       // don't know why this doesn't work
`${StatusMutation.getFragment('viewer')}`       // didn't expect this to work
${Child.getFragment('viewer')}       // didn't expect this to work
`${Child.getFragment('viewer')}`       // didn't expect this to work
```
So whatever this error message means, its suggested fix is incorrect and confusing. Please try to make it more descriptive. Thanks.

**Comment by boosh** [comment]
Actually I think this error message was correct, but I was importing another component that didn't use `getFragment` in this form.

That's one thing with the error messages - it's not very clear exactly where the cause is. I.e. the messages might say *what* is wrong, but you have to hunt around to try to find exactly which bit of code is responsible. Is there some way for error messages to include the file name & line that triggered them, or at least the component name or something?

**Comment by josephsavona** [comment]
@boosh what is the exact error message that you're getting? ``Invariant Violation: RelayQL: Invalid fragment composition, use `${Child.getFragment('name')}`.`` ?

Are you sure that `StatusMutation.getFragment('viewer')` is returning a non-null value?

**Comment by yungsters** [comment]
Ping @boosh.

**Comment by boosh** [comment]
I can't reproduce this now. It was probably fixed by upgrading to 0.3.2, I'm not sure. I'm about to restart work on the client again once I've put the finishing touches to the server. If I run into this issue again I'll reopen this issue with more details.
